app/controllers/chat_channels_controller.rb

```diff
@@ -262,6 +262,31 @@ def change_status
     render json: success_json
   end
 
+  def delete
+    params.require(:chat_channel_id)
+    params.require(:channel_name_confirmation)
+
+    chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
+    raise Discourse::NotFound if chat_channel.blank?
+
+    guardian.ensure_can_delete_chat_channel!
+
+    if chat_channel.name.downcase != params[:channel_name_confirmation].downcase
+      raise Discourse::InvalidParameters
+    end
+
+    chat_channel.trash!(current_user)
+    StaffActionLogger.new(current_user).log_custom(
+      "chat_channel_delete",
+      {
+        chat_channel_id: chat_channel.id,
+        chat_channel_name: chat_channel.name
+      }
+    )
```

Comment:
Perhaps it is safer for this to be in the same DB transaction so we'll never end up with a case where the channel is deleted but no staff action log has been created.

Reply:
Yep good call!